src/target/riscv/batch.c

```diff
@@ -106,7 +107,9 @@ int riscv_batch_run(struct riscv_batch *batch)
 		else
 			jtag_add_dr_scan(batch->target->tap, 1, batch->fields + i, TAP_IDLE);
 
-		if (batch->idle_count > 0)
+		const bool this_scan_resets_delays = resets_delays
+			&& (i >= reset_delays_after_n_scans);
+		if (batch->idle_count > 0 && !this_scan_resets_delays)
```

Comment:
For clarity, I'd recommend to rename `this_scan_resets_delays` to `delays_were_reset` (or similar), which is more descriptive. 

The reason is that this flag becomes `true` for not for just a single scan, but for all scans from some point onwards.

```suggestion
		const bool delays_were_reset = resets_delays
			&& (i >= reset_delays_after_n_scans);
		if (batch->idle_count > 0 && !delays_were_reset)
```

Reply:
Addressed.